app/controllers/recipes_controller.rb

```diff
@@ -46,6 +46,7 @@ def create
 
   # PATCH/PUT /recipes/1 or /recipes/1.json
   def update
+    redirect_to :edit, notice: 'This user is not allowed to update this recipe.' unless authorize @recipe
```

Comment:
This redirect doesn't actually happen because `authorize` raises an error. You can also remove this `authorize` call since `set_recipe` already does it.

You might consider handling the errors as described here https://github.com/varvet/pundit?tab=readme-ov-file#rescuing-a-denied-authorization-in-rails